code/modules/antagonists/changeling/changeling.dm

```diff
@@ -350,12 +350,12 @@
 /datum/antagonist/changeling/proc/regain_powers()
 	emporium_action.Grant(owner.current)
 	for(var/datum/action/changeling/power as anything in innate_powers)
-		power.Grant(owner.current)
+		power.on_purchase(owner.current)
```

Comment:
could be kept to `Grant` if void adaptation applys its things in grant. but I feel then all ling powers should have there things on grant and not purchase